compiler/luci/pass/src/ResolveCustomOpMaxPoolWithArgmaxPass.cpp

```diff
@@ -0,0 +1,814 @@
+  auto cords = graph->nodes()->create<luci::CircleConst>();
+
+  cords->dtype(loco::DataType::FLOAT32);
+
+  cords->rank(4);
+  cords->dim(0).set(1);
+  cords->dim(1).set(output_height);
+  cords->dim(2).set(output_width);
+  cords->dim(3).set(1);
+
+  fill_coords_addition<loco::DataType::FLOAT32>(padding, stride, filter, input_height, input_width,
+                                                cords);
+
+  return cords;
+}
+
+luci::CircleNode *get_custom_output(const luci::CircleCustom *cop, int32_t idx)
+{
+  auto const outputs = loco::succs(cop);
+  assert(outputs.size() == 2);
+
+  auto output = loco::must_cast<luci::CircleCustomOut *>(*outputs.begin());
+  if (output->index() != idx)
+  {
+    output = loco::must_cast<luci::CircleCustomOut *>(*outputs.rbegin());
+  }
+
+  return output;
+}
+
+luci::CircleNode *max_pool_branch(luci::Padding padding, const luci::Stride &stride,
+                                  const luci::Filter filter, luci::CircleCustom *cop)
+{
+  auto graph = cop->graph();
+  auto input = cop->inputs(0);
+
+  auto origin = luci::get_origin(cop);
+  auto name = cop->name() + "/Argmax";
+
+  // Create MaxPool
+  auto maxpool = none_act_func(graph->nodes()->create<luci::CircleMaxPool2D>());
+  {
+    init_name_and_origin(maxpool, name + "/MaxPool2D", origin);
+
+    set_stride(maxpool, stride);
+    set_filter(maxpool, filter);
+    maxpool->padding(padding);
+
+    maxpool->value(input);
+  }
+
+  return maxpool;
+}
+
+luci::CircleNode *window_flattened_coord(const std::string &name, luci::Padding padding,
+                                         const luci::Stride &stride, const luci::Filter filter,
+                                         int32_t input_height, int32_t input_width,
+                                         uint32_t output_height, uint32_t output_width,
+                                         luci::CircleNode *input)
+{
+  auto const graph = input->graph();
+  auto const origin = luci::get_origin(input);
+
+  auto const depth_dimension = 3;
+
+  // Create pad in case of SAME padding
+  luci::CircleNode *conv_input = input;
+  if (padding == luci::Padding::SAME)
+  {
+    auto pad = graph->nodes()->create<luci::CirclePadV2>();
+    init_name_and_origin(pad, name + "/Pad", origin);
+
+    pad->input(input);
+
+    int32_t full_w_pad = compute_full_padding(input_width, output_width, stride.w(), filter.w());
+    int32_t full_h_pad = compute_full_padding(input_height, output_height, stride.h(), filter.h());
+    int32_t left_pad = full_w_pad / 2;
+    int32_t right_pad = full_w_pad - left_pad;
+    int32_t top_pad = full_h_pad / 2;
+    int32_t bottom_pad = full_h_pad - top_pad;
+    auto padding_const = create_padding_const(graph, left_pad, right_pad, top_pad, bottom_pad);
+    init_name_and_origin(padding_const, name + "/Pad_shape", origin);
+    pad->paddings(padding_const);
+
+    auto padding_value =
+      create_scalar<loco::DataType::FLOAT32, float>(graph, std::numeric_limits<float>::lowest());
```

Comment:
@jinevening 
I want to use lowest floating point value to "simulate" max pool with padding bahavior.
These values could be harmful for quantization, how do you feel about filtering such values in record-minmax?